src/main/scala/ie/zalando/fabric/gateway/service/RouteDerivationModels.scala

```diff
@@ -0,0 +1,46 @@
+package ie.zalando.fabric.gateway.service
+
+import ie.zalando.fabric.gateway.models.SynchDomain.{GatewayMeta, GatewaySpec, HttpVerb, PathMatch, RateLimitPeriod}
+
+object RouteDerivationModels {
+
+  // Weights used to avoid collisions from conditional predicates added in Skipper
+  sealed trait WeightedGatewayFeature {
```

Comment:
Why not do this more simply and also dynamically as I suggested in chat:
```
val numberOfPredicates = predicates.count
predicates = predicates ++ weight(numberOfPredicates)
```

Reply:
I was trying to be a little more explicit in the code, so that'd we'd have extra control. Also hadn't gone through the numbers for your suggested change.

Had a look now, detes below:
```
Issue Highlighted:
trafficSwitchAdmin: Host & Traffic & Path & Method & Introspection & Header      			= 6
fallBackAdmin:      Host & Path & Method & Introspection & Header                			= 5
trafficSwitchSvc:   Host & Traffic & Path & Method & Header									= 5
fallBackSvc:        Host & Path & Method & Header									        = 4

Counted Weights:
trafficSwitchAdmin: Host & Traffic & Path & Method & Introspection & Header & Weight(4)     = 10
fallBackAdmin:      Host & Path & Method & Introspection & Header & Weight(4)               = 9
trafficSwitchSvc:   Host & Traffic & Path & Method & Header & Weight(3)                     = 8
fallBackSvc:        Host & Path & Method & Header & Weight(3)								= 7
tsGlobalRateLimit:  Host & Traffic & Path & Method & Header & Weight(3)                     = 8
fbGlobalRateLimit:  Host & Path & Method & Header & Weight(3)                     			= 7
tsSvcRateLimit:     Host & Traffic & Path & Method & Introspection & Header & Weight(4)       = 10
fbSvcRateLimit:     Host & Path & Method & Introspection & Header & Weight(4)       		= 9

Gapped Weights:
trafficSwitchAdmin: Host & Traffic & Path & Method & Introspection & Header & Weight(100)   = 106
fallBackAdmin:      Host & Path & Method & Introspection & Header & Weight(100)             = 105
trafficSwitchSvc:   Host & Traffic & Path & Method & Header & Weight(40)                    = 45
fallBackSvc:        Host & Path & Method & Header & Weight(40)								= 44
tsGlobalRateLimit:  Host & Traffic & Path & Method & Header & Weight(60)                    = 65
fbGlobalRateLimit:  Host & Path & Method & Header & Weight(60)                     			= 64
tsSvcRateLimit:     Host & Traffic & Path & Method & Introspection & Header & Weight(80)    = 86
fbSvcRateLimit:     Host & Path & Method & Introspection & Header & Weight(80)       		= 85
```

I think the only collisions, probably can't happen (i.e. normal svc route and rate limited svc route shouldnt be able to exist together). I can change it if ya want, but I think it probably relies on more implicit knowledge to understand the shorter change...

Reply:
The approach I suggest just widens the gap between each of the existing predicate sets to allow Skipper to insert a traffic predicate safely

Reply:
Yup, they both ultimately do the same thing. I thought this way was more descriptive from a code point of view though. Happy to change though if you don't think there;s any advantage offsetting the extra verbosity.

Reply:
What about the default catch-all routes with 1 and 2 predicates, are they not affected?
Seems safer to just iterate through all routes and ensure the predicate gaps jump in 2s rather than 1s

Reply:
Yeah, I vote for less verbosity.